workspaces/ui-v2/src/lib/shape-diffs/build-inner-shape.ts

```diff
@@ -40,16 +41,21 @@ export function builderInnerShapeFromChoices(
       return foundShapeId[0];
     } else {
       const filterToTarget = actual.trailAffordances.map((affordance) => {
-        return {
-          ...affordance,
-          wasString: i === ICoreShapeKinds.StringKind,
-          wasNumber: i === ICoreShapeKinds.NumberKind,
-          wasBoolean: i === ICoreShapeKinds.BooleanKind,
-          wasNull: i === ICoreShapeKinds.NullableKind,
-          wasArray: i === ICoreShapeKinds.ListKind,
-          wasObject: i === ICoreShapeKinds.ObjectKind,
-          fieldSet: i === ICoreShapeKinds.ObjectKind ? affordance.fieldSet : [],
-        };
+        if (equals(affordance.trail, actual.jsonTrail)) {
```

Comment:
It will create the commands if you pass in the affordances, I guess that's a limitation of this new approach to use sort order to get dependencies right. Can the UI reason about only including children affordances for objects and arrays?

Reply:
Looking at the domain logic, perhaps we _can_ actually choose to filter out shape prototypes that don't have a parent that's attached to the root, as we are keeping generating them by trails (and thus can use trail arithmetic). By filtering out those shape prototypes, their commands won't be generated.

Without it, the use case you mention will add orphaned shapes to the specification log, which is probably something we'd want to avoid.